### PR TITLE
Run tests using shaded dependencies

### PR DESCRIPTION
**Pull request by bsideup**

### CODE REVIEW

File: modules/k3s/build.gradle [hunk 1]
```diff
@@ -3,7 +3,8 @@ description = "Testcontainers :: K3S"
 dependencies {
     api project(":testcontainers")
 
-    shaded 'com.fasterxml.jackson.dataformat:jackson-dataformat-yaml:2.13.1'
+    // https://youtu.be/otCpCn0l4Wo
```
Comment (kiview) [inline modules/k3s/build.gradle:6]: lol 🤣

**Comment by kiview** [comment]
Did we end up with the wrong docker-java version through this?
https://ge.testcontainers.org/s/sp5tcx75pjjyg/tests/overview?outcome=failed

**Comment by rnorth** [comment]
> Did we end up with the wrong docker-java version through this? [ge.testcontainers.org/s/sp5tcx75pjjyg/tests/overview?outcome=failed](https://ge.testcontainers.org/s/sp5tcx75pjjyg/tests/overview?outcome=failed)

I don't know, but another possibility is that we've found two tests that should have been failing but haven't been...

**Comment by bsideup** [comment]
@kiview @rnorth these two are failing because of the class mismatch - shaded in TC, but tests are using the non-shaded versions. I have the fix locally, 1 sec

**Review by kiview — APPROVED** [review]

File: modules/k3s/build.gradle (hunk 1 above)
Comment (rnorth) [reply modules/k3s/build.gradle:6]: Maybe we should say why, and under what circumstances this should be changed.

Have we disabled dependabot for this module/dependency combination yet?

**Comment by rnorth** [comment]
Ping @YannickWeber for awareness
We need to set up CODEOWNERS some time...

@bsideup perhaps the title of this PR should be tweaked and/or the description modified, so that we or others can understand why this was done in 5 years time. Maybe a title like `Run Testcontainers unit tests using shaded dependencies` or something?

File: modules/k3s/build.gradle (hunk 1 above)
Comment (bsideup) [reply modules/k3s/build.gradle:6]: done! 👍

Comment (bsideup) [reply modules/k3s/build.gradle:6]: We can keep Dependabot as the CI would fail now

**Comment by bsideup** [comment]
@rnorth good idea, changed the title 👍

**Review by rnorth — APPROVED** [review]

**Pull request merged**